lamindb_setup/dev/cloud_sqlite_locker.py

```diff
@@ -190,3 +197,26 @@ def get_locker(isettings: InstanceSettings) -> Locker:
         _locker = Locker(user_id, storage_root, instance_name)
 
     return _locker
+
+
+# decorator
+def unlock_on_exception(func):
+    # https://stackoverflow.com/questions/1782843/python-decorator-handling-docstrings
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except Exception as exc:
+            if isinstance(exc, InstanceLockedException):
+                raise exc
+            # this is ugly but i don't see any other way to do it properly
+            # this is ignored because if not then the current instance could be unlocked
+            # during the try of loading a new instance
+            elif isinstance(exc, RuntimeError):
+                if str(exc).startswith("Currently don't support init or load"):
```

Comment:
This is dangerous as it might easily change in the future.

Reply:
Changed logic there, removed this particular case and added the parameter that allows to ignore exceptions if locker is the same as the previous locker after the wrapped function execution.